.github/workflows/website.yml

```diff
@@ -0,0 +1,156 @@
+name: Website
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["*"]
+
+jobs:
+  docs-home:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          lfs: true
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: yarn
+
+      - run: corepack enable && yarn install --immutable
+
+      - run: yarn build
+        working-directory: website
+
+      - run: yarn typedoc
+
+      - run: |
+          mkdir -p website/build/docs
+          mv __docs__/typescript website/build/docs/typescript
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: public
+          path: website/build
+          retention-days: 1
+
+  docs-cpp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: make ci-docs
+        working-directory: cpp
+
+      - run: tar -czf docs-cpp.tgz __docs__/cpp
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: docs-cpp
+          path: docs-cpp.tgz
+          retention-days: 1
+
+  docs-python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+
+      - run: pip install pipenv
+
+      - run: make -C python docs
+
+      - run: tar -czf docs-python.tgz __docs__/python
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: docs-python
+          path: docs-python.tgz
+          retention-days: 1
+
+  docs-swift:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: swift-actions/setup-swift@v1
+        with:
+          swift-version: "5.7"
+
+      - run: mkdir __docs__
+
+      - run: |
+          swift package \
+            --allow-writing-to-directory __docs__ \
+            generate-documentation --target MCAP \
+            --output-path __docs__/swift \
+            --disable-indexing \
+            --transform-for-static-hosting \
+            --hosting-base-path docs/swift
+
+      # https://github.com/actions/upload-artifact/issues/85
+      - run: tar -czf docs-swift.tgz __docs__/swift
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: docs-swift
+          path: docs-swift.tgz
+          retention-days: 1
+
+  docs-rust:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+
+      - run: cargo doc --no-deps
+        working-directory: rust
+
+      - run: mkdir __docs__
+
+      - run: cp -r rust/target/doc __docs__/rust
+
+      - run: tar -czf docs-rust.tgz __docs__/rust
+      - uses: actions/upload-artifact@v3
+        with:
+          name: docs-rust
+          path: docs-rust.tgz
+
+  docs-deploy:
+    runs-on: ubuntu-latest
+    environment:
+      name: mcap (Preview)
```

Comment:
What deploys this to production once it merges to main?

Reply:
I'm not super happy with this but would like to figure it out in a separate PR. This whole `environment` section is not needed and the only thing it seems to do is make the "view deployment" button appear on the PR. The actual deployment happens regardless.

I think probably the upstream cloudflare action is missing something to cause the PR deployment status to show up.

Reply:
Also the `deploy` step has an output called `steps.deploy.outputs.environment` with the environment name but Github does not let you use variables in this `name:` field.

Anyway like I said this line is purely cosmetic and I don't think should block merging the PR.

Reply:
Turns out this is fixed in the latest version of the pages action.